File: languages/go/tools/test/PrintAST/Example1.go
// Example1 test case for PrintAST.ql
package main

import (
	"fmt"
	"os"
)

type Example1 struct {
	name string
}

func NewExample1(name string) *Example1 {
	if name == "" {
		name = "World"
	}
	return &Example1{name: name}
}

func (e *Example1) demo() {
	numbers := []int{1, 2, 3}

	// For loop
	for i := 0; i < len(numbers); i++ {
		fmt.Println(numbers[i])
	}

	// Range loop
	for _, n := range numbers {
		fmt.Println(n * 2)
	}

	// Struct and string formatting
	person := struct {
		name string
		age  int
	}{e.name, 25}
	fmt.Printf("Hello, %s!\n", person.name)

	// Error handling with defer
	defer func() {
		if r := recover(); r != nil {
			fmt.Println("Error occurred")
		}
	}()
}

func helper() string {
	return "Helper"
}

func main() {
	// $Example1

	example := NewExample1("")
	message := "Hello World!"

	if len(os.Args) > 1 {
		example.name = os.Args[1]
	} else {
		fmt.Println(message)
	}

	example.demo()
}

File: languages/java/tools/test/PrintAST/Example1.java
// Example1 test case for PrintAST.ql

import java.util.Arrays;
import java.util.List;

public class Example1 {
    private String name;
    
    public Example1() {
        this("World");
    }
    
    public Example1(String name) {
        this.name = name;
    }
    
    public static void main(String[] args) {
        // $Example1
        
        Example1 example = new Example1();
        String message = "Hello World!";
        
        if (args.length > 0) {
            example.name = args[0];
        } else {
            System.out.println(message);
        }
        
        example.demo();
    }
    
    public void demo() {
        List<Integer> numbers = Arrays.asList(1, 2, 3);
        
        // For loop
        for (int i = 0; i < numbers.size(); i++) {
            System.out.println(numbers.get(i));
        }
        
        // Enhanced for loop
        for (int n : numbers) {
            System.out.println(n * 2);
        }
        
        // Stream with lambda
        numbers.stream().forEach(n -> System.out.println(n * 3));
        
        // Object creation and string formatting
        Person person = new Person(this.name, 25);
        System.out.printf("Hello, %s (age %d)!%n", person.getName(), person.getAge());
        
        // Try-catch
        try {
            int result = 10 / 0;
            System.out.println(result);
        } catch (ArithmeticException e) {
            System.out.println("Error occurred");
        }
    }
    
    private static class Person {
        private String name;
        private int age;
        
        public Person(String name, int age) {
            this.name = name;
            this.age = age;
        }
        
        public String getName() {
            return name;
        }
        
        public int getAge() {
            return age;
        }
    }
    
    public static String helper() {
        return "Helper";
    }
}
